src/EscapeCodeGenerator.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Defs.hpp"
+#include <string>
+
+struct EscapeCodeGenerator {
+
+    using EscapeCode = std::string;
+
+    virtual EscapeCode cursorForward(int) { return EscapeCode{}; }
+    virtual EscapeCode cursorBackward(int) { return EscapeCode{}; }
+    virtual EscapeCode clearTerminalLine() { return EscapeCode{}; }
+    virtual EscapeCode setColor(Color) { return EscapeCode{}; }
+
+};
+
+struct ANSIEscapeCodeGenerator : public EscapeCodeGenerator {
+
+    EscapeCode cursorForward(int n) override {
```

Comment:
is it possible to have negative number here? if not, then please switch to undefined for better type-safety